chat-widget/src/components/livechatwidget/common/startChat.ts

```diff
@@ -228,4 +237,47 @@ const setCustomContextParams = (chatSDK: any) => {
     }
 };
 
-export { prepareStartChat, initStartChat, setPreChatAndInitiateChat };
\ No newline at end of file
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const checkIfConversationStillValid = async (chatSDK: any, props: any, requestId: any): Promise<boolean> => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let conversationDetails: any = undefined;
+
+    //For auth chat
+    if (props.getAuthToken) {
+        const authClientFunction = getAuthClientFunction(props.chatConfig);
+        if (props.getAuthToken && authClientFunction) {
+            // set auth token to chat sdk before start chat
+            const authSuccess = await handleAuthentication(chatSDK, props.chatConfig, props.getAuthToken);
+            if (!authSuccess) {
+                return false;
+            }
+        }
+    }
+
+    //Preserve old requestId
+    const oldRequestId = chatSDK.requestId;
```

Comment:
could we considerate the idea of breaking this function in 2?, aiming to have atomic functions that have a single responsability, instead of large complex functions.

this will help for readability, debugging,  maintenance...

Reply:
This function has only one job to find valid conversation. If you meant by auth part, its needed for auth chats as same function is used for both chats. Anyhow, I moved the auth part to a functions.